### ISSUE TITLE
systemd-logind causing multiple connections to /dev/dri/card0 with xorg

### ISSUE THREAD

**Issue by xogium** [systemd/systemd#3782]
### Submission type
- [x] Bug report
- [ ] Request for enhancement (RFE)

_NOTE: Do not submit anything other than bug reports or RFEs via the issue tracker!_
### systemd version the issue has been seen with

> systemd 230

_NOTE: Do not submit bug reports about anything but the two most recently released systemd versions upstream!_
### Used distribution

> ArchLinux
### Expected behaviour you didn't see

> systemd-logind should open the file for i/o for xorg only once, and resume the old connection if x is resuming from suspend. Or close the old connection and reopen a new one
### Unexpected behaviour you saw

> systemd-logind is causing /dev/dri/card0 to be reopened each time X is resuming from suspend (i.e: switching from another tty back to X), eventually leading to the impossibility to start any X application. fd gets more and more connections, till it can't handles it.
### Steps to reproduce the problem

> 1. start X and let systemd controls the session. With a display manager or a simple startx command.
> 2. Log in to another tty (another X server or a pure console will both do).
> 3. Switch from the tty where X is running to the other tty, and monitor with 'lsof -p $(pidof Xorg) | wc -l'. See that the number is slowly increasing. That will eventually leads to a 'maximum number of clients reached' error.
### Aditionnal informations

> Here, I put the two references I have for that bug. One is a bug opened on bugzilla against xorg itself, which was later proved to be a systemd-logind issue (see last comment). The other is a post on archlinux forums to 'fix' that by no longer letting systemd-logind in control of the session.
#### ArchLinux forum post

> https://bbs.archlinux.org/viewtopic.php?id=190272
#### bug report against xorg on bugzilla

> https://bugs.freedesktop.org/show_bug.cgi?id=86665

**Comment by keszybz** [comment]
I don't understand: the last comment on the bug you linked to is a patch for Xorg to close the fd (https://bugs.freedesktop.org/show_bug.cgi?id=86665#c6). I haven't looked at the details, but it seems reasonable that if the problem is that Xorg keeps multiple references to the card open, the bug is in Xorg, not in systemd-logind. Please explain why you think systemd-logind is too blame.

**Comment by xogium** [comment]
Hi, and thanks for the quick reply :)
I'm sory if this isn't a systemd bug. Is just that this bug has been present for a really long time (since 2014 from what I could tell from the date on both references I have). The last message on bugzilla might have me thinking it was a systemd bug instead of a Xorg one because of that line:
diff --git a/hw/xfree86/os-support/linux/systemd-logind.c b/hw/xfree86/os-support/linux/systemd-logind.c

So that's why I thought the issue would be more a systemd one, plus the topic I found on archlinux forum. As I said, if that was an issue opened by error, let me know and I'll close it. I just want that two years old bug to be fixed, and I'm getting kind of desperate to get it fixed since there's no comment on that bugzilla report for a while. And even if there had some updates to xorg in archlinux, the problem wasn't fixed yet. So I'm confused. Is this a systemd bug, or a xorg one ? :)
Thanks again

**Comment by webczat** [comment]
It is related to logind only because logind mediates, do I understand
this right? access to devices like to a graphic card. but it is xorg who
requests that access and who opens and uses the file descriptor.

**Comment by xogium** [comment]
Hi,
yes I think I get that right... I really got comfused on that one, sorry about that :)
Let's hope it'll be fixed one day. I tried the 'fix' on the arch forum and after x would only run as root. The problem was fixed, but I didn't like that fact, so I went back to letting systemd-logind in control, since my screen reader was started but didn't want to speak as root.
I guess I should be closing this now.. Anyway, thanks :)
